platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md

```diff
@@ -0,0 +1,203 @@
+# Durably Persist Self-Events Before Gossip
+
+---
+
+## Summary
+
+If a self-event is gossiped and the node is shutdown before the self-event is persisted to disk, then the node will
+branch in the hashgraph on restart by building a new self-event on the same self-event parent it had previously used.
+In other words, _a node must only tell the world of a self-event **after** it has recorded this information!_
+
+This proposal ensures self-events are persisted to disk before they are gossiped. It simplifies the relationship
+between the consensus engine and the PCES writer, removing the need for the PCES sequencer, the `RoundDurabilityBuffer`,
+and the responsibility of the consensus engine to tell the PCES writer when to flush, while introducing negligible
+additional latency.
+
+| Metadata           | Entities                            | 
+|--------------------|-------------------------------------|
+| Designers          | Cody, Richard, Jasper               |
+| Functional Impacts | Platform                            |
+| Related Proposals  | N/A                                 |
+| HIPS               | N/A                                 |
+
+---
+
+## Purpose and Context
+
+Nodes in the network create new "self-events" on a regular basis. These self-events carry transactions that were
```

Comment:
The clear explanation of the mechanism of branching is very appreciated.

Reply:
indeed! Learning a ton from reading the design docs so far. Yay, design!